pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -342,6 +349,23 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	// Set region label on the ClusterDeployment
+	if region := getClusterRegion(cd); cd.Spec.Platform.BareMetal == nil && cd.Labels[hivev1.HiveClusterRegionLabel] != region {
```

Comment:
I don't think we need to check BareMetal is nil, just getClusterRegion, see if the label is not what we expect. If expected is "unknown" that's fine.

Reply:
I disagree. I do not think that we should be adding a region label for platforms where region is not a concept.